File: day-01/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)

func TopThreeCalories(calorieList []string) int {
	totalElves := 0
	curCals := 0
	var totalCals []int
	for _, cals := range calorieList {
		if cals == "" {
			totalCals = append(totalCals, curCals)
			totalElves++
			curCals = 0
		} else {
			int_cals, err := strconv.Atoi(cals)
			if err != nil {
				panic(err)
			}
			curCals += int_cals
		}
	}
	totalCals = append(totalCals, curCals)
	sort.Ints(totalCals)
	return totalCals[totalElves] + totalCals[totalElves-1] + totalCals[totalElves-2]
}

func TopCalories(calorieList []string) int {
	curCals := 0
	topCals := 0
	for _, cals := range calorieList {
		if cals == "" {
			// Check for top status
			if curCals > topCals {
				topCals = curCals
			}
			// Next elf up
			curCals = 0
		} else {
			int_cals, err := strconv.Atoi(cals)
			if err != nil {
				panic(err)
			}
			curCals += int_cals
		}
	}
	if curCals > topCals {
		topCals = curCals
	}
	return topCals
}

func main() {
	readFile, err := os.Open("input.txt")
	if err != nil {
		fmt.Println(err)
	}
	defer readFile.Close()

	fileScanner := bufio.NewScanner(readFile)
	fileScanner.Split(bufio.ScanLines)

	var fileLines []string

	for fileScanner.Scan() {
		fileLines = append(fileLines, fileScanner.Text())
	}

	topCals := TopCalories(fileLines)
	topThreeCals := TopThreeCalories(fileLines)

	fmt.Printf("The top elf has a calorie count of %v\n", topCals)
	fmt.Printf("The top three elves have a calorie count of %v\n", topThreeCals)

}

File: day-01/main.py
import os

def top_calories(input_lines):
    top_calories = 0
    cur_calories = 0
    for line in input_lines:
        if line == "":
            if cur_calories > top_calories:
                top_calories = cur_calories
            cur_calories = 0
        else:
            cur_calories += int(line)
    if cur_calories > top_calories:
        top_calories = cur_calories
    return top_calories


def top_three_calories(input_lines):
    top_calories = []
    cur_calories = 0
    for line in input_lines:
        if line == "":
            top_calories.append(cur_calories)
            cur_calories = 0
        else:
            cur_calories += int(line)
    top_calories.append(cur_calories)
    top_calories.sort(reverse=True)
    return sum(top_calories[0:3])

if __name__ == '__main__':
    with open('input.txt') as in_fh:
        file_lines = [line.strip() for line in in_fh.readlines()]

    top_cals = top_calories(file_lines)
    top_three_cals = top_three_calories(file_lines)
    print(f"Top calories held: {top_cals}")
    print(f"Top three calories held: {top_three_cals}")
